workloads/imagenet/imagenet_jax/workload.py

```diff
@@ -0,0 +1,263 @@
+
+  @property
+  def loss_type(self):
+    return spec.LossType.SOFTMAX_CROSS_ENTROPY
+
+  @property
+  def train_mean(self):
+    return [0.485 * 255, 0.456 * 255, 0.406 * 255]
+
+  @property
+  def train_stddev(self):
+    return [0.229 * 255, 0.224 * 255, 0.225 * 255]
+
+  @property
+  def num_train_examples(self):
+    if 'imagenet2012' in self.dataset:
+      return 1271167
+    if 'imagenette' == self.dataset:
+      return 9469
+
+  @property
+  def num_eval_examples(self):
+    if 'imagenet2012' in self.dataset:
+      return 100000
+    if 'imagenette' == self.dataset:
+      return 3925
+
+  def model_params_types(self):
+    pass
+
+  @property
+  def max_allowed_runtime_sec(self):
+    return 111600 # 31 hours
+
+  @property
+  def eval_period_time_sec(self):
+    return 6000 # 100 mins
+
+  # Return whether or not a key in spec.ParameterTree is the output layer
+  # parameters.
+  def is_output_params(self, param_key: spec.ParameterKey) -> bool:
+    pass
+
+  def _build_dataset(self,
+      data_rng: jax.random.PRNGKey,
+      split: str,
+      data_dir: str,
+      batch_size):
+    if batch_size % jax.device_count() > 0:
+      raise ValueError('Batch size must be divisible by the number of devices')
+    ds_builder = tfds.builder(self.dataset)
+    ds_builder.download_and_prepare()
+    ds = input_pipeline.create_input_iter(
+      ds_builder,
+      batch_size,
+      self.train_mean,
+      self.train_stddev,
+      train=True,
+      cache=False)
+    return ds
+
+  def build_input_queue(
+      self,
+      data_rng: jax.random.PRNGKey,
+      split: str,
+      data_dir: str,
+      batch_size: int):
+    return iter(self._build_dataset(data_rng, split, data_dir, batch_size))
+
+  def sync_batch_stats(self, model_state):
+    """Sync the batch statistics across replicas."""
+    # An axis_name is passed to pmap which can then be used by pmean.
+    # In this case each device has its own version of the batch statistics and
+    # we average them.
+    avg_fn = jax.pmap(lambda x: lax.pmean(x, 'x'), 'x')
+    new_model_state = model_state.copy({
+      'batch_stats': avg_fn(model_state['batch_stats'])})
+    return new_model_state
+
+  @property
+  def param_shapes(self):
+    if self._param_shapes is None:
+      raise ValueError('This should not happen, workload.init_model_fn() '
+                       'should be called before workload.param_shapes!')
+    return self._param_shapes
+
+  def preprocess_for_train(
+      self,
+      selected_raw_input_batch: spec.Tensor,
+      selected_label_batch: spec.Tensor,
+      train_mean: spec.Tensor,
+      train_stddev: spec.Tensor,
+      rng: spec.RandomState) -> spec.Tensor:
+    return (selected_raw_input_batch, selected_label_batch)
+
+  def preprocess_for_eval(
+      self,
+      raw_input_batch: spec.Tensor,
+      raw_label_batch: spec.Tensor,
+      train_mean: spec.Tensor,
+      train_stddev: spec.Tensor) -> spec.Tensor:
+    return (raw_input_batch, raw_label_batch)
```

Comment:
we can delete these two `preprocess_for_train` and `preprocess_for_eval` functions

Reply:
These are functions in the `spec.py`. Should keep these functions?

```
$ git grep -n -e preprocess_for_eval -e preprocess_for_train
spec.py:154:  def preprocess_for_train(
spec.py:164:  def preprocess_for_eval(
```

Reply:
Ah yeah go ahead and delete those too, also could you delete them in the [MNIST example](https://github.com/mlcommons/algorithmic-efficiency/blob/main/workloads/mnist/mnist_jax/workload.py#L81) and edit the [submission runner](https://github.com/mlcommons/algorithmic-efficiency/blob/main/submission_runner.py#L175) to get rid of them too (just use `selected_train_{input,label}_batch` below).

Reply:
Ok. In [my commit](https://github.com/mlcommons/algorithmic-efficiency/pull/11/commits/3d5e909c58e77365d1f4d4c20cd027ada3d33ad4) I deleted the preprocess furctions from the imagenet workload, spec, and submission_runner but I didn't delete it for MNIST (jax or pytorch implementations) because the [pytorch MNIST actually does preprocessing](https://github.com/mlcommons/algorithmic-efficiency/blob/main/workloads/mnist/mnist_pytorch/workload.py#L109) and the [common MNIST workload uses it](https://github.com/mlcommons/algorithmic-efficiency/blob/main/workloads/mnist/workload.py#L53). It would be nice to apply that pytorch-only preprocessing to at `build_input_queue()` time, but I did not spend enough time to find a way that worked in pytorch. Maybe @hanlint knows. So in my commit for this, I didn't change MNIST usage of preprocess functions, though, I had to [added one preprocess call](https://github.com/mlcommons/algorithmic-efficiency/pull/11/commits/3d5e909c58e77365d1f4d4c20cd027ada3d33ad4#diff-3cf4126d2419ed538704e070b881b435f2d2c9b0ef833cc2583921833e45e450R50) to make up for the removal from submission_runner.py.

I also renamed `augmented_and_preprocessed_input_batch` to `input_batch` because we aren't mandating a preprocessing function call anymore, so it doesn't make sense to always have that name. The spec was requiring it.